envs/block_picking_env.py

```diff
@@ -10,27 +10,40 @@ class BlockPickingEnv(simulator_base_env):
     def __init__(self, config):
       if simulator_base_env is NumpyEnv:
         super(BlockPickingEnv, self).__init__(config['seed'], config['workspace'], config['max_steps'],
-                                              config['obs_size'], config['render'])
+                                              config['obs_size'], config['render'], config['action_sequence'])
       elif simulator_base_env is VrepEnv:
         super(BlockPickingEnv, self).__init__(config['seed'], config['workspace'], config['max_steps'],
-                                              config['obs_size'], config['port'], config['fast_mode'])
+                                              config['obs_size'], config['port'], config['fast_mode'],
+                                              config['action_sequence'])
       elif simulator_base_env is PyBulletEnv:
         super(BlockPickingEnv, self).__init__(config['seed'], config['workspace'], config['max_steps'],
-                                              config['obs_size'], config['fast_mode'], config['render'])
+                                              config['obs_size'], config['fast_mode'], config['render'],
+                                              config['action_sequence'])
       else:
         raise ValueError('Bad simulator base env specified.')
+      self.simulator_base_env = simulator_base_env
 
     def reset(self):
       ''''''
       super(BlockPickingEnv, self).reset()
 
-      self.block = self._generateObjects(0, 1)[0]
+      self.block = self._generateShapes(0, 1, random_orientation=True)[0]
 
       return self._getObservation()
 
     def _checkTermination(self):
       ''''''
-      return self._isObjectHeld(self.block)
```

Comment:
Can you pull the code here for PyBullet/Vrep into a function inside their base envs with the same name as the numpy env so this call will just go back to: return slf.isHolding()

Reply:
Please the new version